examples/ui/drag_drop.rs

```diff
@@ -0,0 +1,374 @@
+
+#[derive(Component)]
+struct SourceDropContainer(pub Entity);
+
+#[derive(Component)]
+pub struct DropContainer(bool);
+
+impl Default for DropContainer {
+    fn default() -> Self {
+        DropContainer(true)
+    }
+}
+
+struct DragableHoverEvent(pub Entity);
+struct DragableClickEvent(pub Entity);
+struct DragableDropEvent(pub Entity);
+
+#[derive(Component)]
+struct MouseIn;
+
+struct MouseEnterEvent(pub Entity);
+struct MouseExitEvent(pub Entity);
+
+struct DragDropPlugin;
+
+pub const CLEAR_COLOR: Color = Color::rgb(0.1, 0.1, 0.1);
+
+impl Plugin for DragDropPlugin {
+    fn build(&self, app: &mut App) {
+        app.insert_resource(ImageSettings::default_nearest())
+            .insert_resource(ClearColor(CLEAR_COLOR))
+            .add_plugins(DefaultPlugins)
+            .add_startup_system_to_stage(StartupStage::PreStartup, spawn_camera)
+            .add_startup_system_to_stage(StartupStage::PostStartup, spawn_scene)
+            .add_event::<DragableClickEvent>()
+            .add_event::<DragableHoverEvent>()
+            .add_event::<DragableDropEvent>()
+            .add_event::<MouseEnterEvent>()
+            .add_event::<MouseExitEvent>()
+            // System order is important
+            .add_system(mouse_drag_active_moving)
+            .add_system(mouse_drag_active_click)
+            .add_system(mouse_drag_not_active)
+            .add_system(mouse_hover)
+            .add_system(mouse_exit.after(mouse_hover))
+            .add_system(mouse_enter.after(mouse_exit))
+            .add_system(dragable_drop_event.after(mouse_drag_active_click))
+            .add_system(dragable_click_event.after(mouse_drag_not_active))
+            .add_system(dragable_starting.after(dragable_click_event));
+    }
+}
+
+fn main() {
+    App::new().add_plugin(DragDropPlugin).run();
+}
+
+fn spawn_camera(mut commands: Commands) {
+    commands.spawn_bundle(Camera2dBundle::default());
+}
+
+fn spawn_scene(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands
+        .spawn_bundle(NodeBundle {
+            style: Style {
+                size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
+                justify_content: JustifyContent::SpaceBetween,
+                flex_direction: FlexDirection::Column,
+                ..default()
+            },
+            color: Color::NONE.into(),
+            ..default()
+        })
+        .with_children(|parent| {
+            parent
+                .spawn_bundle(NodeBundle {
+                    style: Style {
+                        size: Size::new(Val::Percent(100.), Val::Px(225.0)),
+                        border: UiRect::all(Val::Px(2.0)),
+                        ..default()
+                    },
+                    color: Color::rgba(1., 1., 1., 0.01).into(),
+                    ..default()
+                })
+                .insert(Interaction::default())
+                .insert(FocusPolicy::default())
+                .insert(DropContainer::default())
```

Comment:
Interaction + FocusPolicy + DropContainer feels like it should be a bundle type :)

Reply:
Great idea